src/utils/miscellaneous.ts

```diff
@@ -20,18 +20,32 @@ export const log = process.env.NODE_ENV === 'test' ? noop : console.log
 
 export function getToken<T extends TokenDetails, K extends keyof T>(
   key: K,
-  value: string | undefined = '',
+  value: string | number | undefined = '',
   tokens: T[] | undefined | null,
 ): T | undefined {
-  const valueUppercase = value.toUpperCase()
   return (tokens || []).find((token: T): boolean => {
     const tokenKeyValue = token[key]
     if (tokenKeyValue) {
-      return tokenKeyValue?.['toString']().toUpperCase() === valueUppercase
+      switch (typeof tokenKeyValue) {
+        case 'string':
+          return tokenKeyValue?.['toString']().toUpperCase() === (value as string).toUpperCase()
```

Comment:
why `toString()` if it's already string? TS complains?

Reply:
why get rid of `const valueUppercase = value.toUpperCase()`?

Reply:
yep, `valueUppercase` was done so we don't convert several times the same value to upercase.

Reply:
> why toString() if it's already string? TS complains?

I guess it did complain before these change. Now it doesn't, will fix it.

> const valueUppercase = value.toUpperCase()?

Because the value might be a number, thus doing `.toUpperCase()` only when string inside the switch case.